core/src/smartcontracts/isi/world.rs

```diff
@@ -174,6 +174,66 @@ pub mod isi {
         }
     }
 
+    impl Execute for Grant<PermissionToken, Role> {
```

Comment:
```suggestion
    impl Execute for Mint<PermissionToken, Role> {
```

I think this would be more appropriate

Reply:
not sure, we used `Grant/Revoke` for everything permission related before

Reply:
Actually, distinction between between this instructructions is very vague.
And in both cases this instructions gather really different user cases under the same roof.

Reply:
> Actually, distinction between between this instructructions is very vague.
And in both cases this instructions gather really different user cases under the same roof.

yes, that's why we have #3924 but until then:

IMO you can Grant/Revoke a permission to an account, but Mint/Burn a permission for a role. Grant implies that the one receiving is some sort of an actor and is then able to perform additional actions after the grant. Mint has a connotation of just adding to the previously existing value of/collection of items.

I'd like to get more opinions here. @0x009922

Reply:
Either is acceptable to me.
"Mint" can be interpreted in various ways, as there are already `Mint<PublicKey, Account>` etc.
On the other hand, permission could be "granted" to role as an abstract actor or something like trait over accounts.

The only difference as a type seems to me to whether the destination is generic or not.
I'm not sure if there they need to be differentiated in the data model